### PR TITLE
fix: fix social media and sharing icons

### PR DESCRIPTION
**Pull request by justinhynes**
The Credentials IDA gives our users the ability to share or print their program certificates if they wish. This functionality is enabled by rendering icons (provided via a third party package called `fontawesome`).

Recently, something caused this functionality to break and the icons are rendering as squares.

This PR attempts to fix this issue. Initially, we attempted to troubleshoot the original issue with the old `fontawesome` package (which was released 7 years ago). We were unable to do that. Then, we tried to update to the latest version of fontawesome via NPM. This caused additional issues that weren't well understood. Finally, I have decided to remove the npm package in favor of a pip package that is designed to work specifically with Django.

Thus, this PR adds a new default installed app to the IDA's `installed_apps` setting -- `fontawesomefree`.

This approach worked and has fixed the display of the icons used in the sharing options for a program certificate rendered by the Credentials IDA.

Additionally, an update was made to the `base.html` template to import the fontawesome icons, as well as an update to the `_actions.html` template to update the class names appropriate to the latest version of `fontawesome`.

Before:
image

After:
image

**Run JavaScript tests locally with Karma**

There is work being done on a fix to get Karma to run in CI. Until then, however, contributors are required to run these tests locally.

- [x] Make sure you are inside the devstack container
- [x] Run `make test-karma`
- [x] All tests pass

### CODE REVIEW

**Review by deborahgu — APPROVED** [review]
this looks good to me once the conflicts resolve.

File: credentials/templates/_actions.html [hunk 1]
```diff
@@ -25,7 +25,7 @@ <h3 class="sr-only">{% trans 'Print or share your certificate' as tmsg %}{{ tmsg
               data-track-event-property-program-uuid="{{ user_credential.credential.program_uuid }}">
         <a class="share-link" target="_blank"
            href="https://twitter.com/intent/tweet?text={{ tweet_text|urlencode }}&url={{ share_url|urlencode }}{% if twitter_username %}&via={{ twitter_username }}{% endif %}">
-          <span class="fa fa-twitter" aria-hidden="true"></span>
+          <span class="fa-brands fa-twitter" aria-hidden="true"></span>
```
Comment (justinhynes) [inline credentials/templates/_actions.html:28]: classnames changed from fontawesome v4.x --> v.6x, brand icons require the `fa-brands` classname now.

File: credentials/settings/base.py [hunk 2]
```diff
@@ -63,6 +63,7 @@
     "drf_yasg",
     "xss_utils",
     "openedx_events",
+    "fontawesomefree",
```
Comment (justinhynes) [inline credentials/settings/base.py:66]: adding `fontawesomefree` as a new third party app for the Credentials IDA

File: requirements/base.txt [hunk 3]
```diff
@@ -174,7 +176,7 @@ jinja2==3.1.3
     # via
     #   code-annotations
     #   coreschema
-lxml[html-clean,html_clean]==5.1.1
+lxml[html-clean]==5.1.1
```
Comment (justinhynes) [inline requirements/base.txt:179]: this duplication looked fishy, so I removed one of them

Comment (deborahgu) [reply requirements/base.txt:179]: as I recall, when I added the first one to `base.in`, the make command that builds requirements added the 2nd.  Should we be manually editing `base.txt`?  I thought it was always autogenerated.

Comment (justinhynes) [reply requirements/base.txt:179]: I may have made a bad assumption. I was fixing up some conflicts and thought this was part of my issues. Generally I'm against modifying the generated files, but maybe I'll just run through another `make upgrade` and use what it gives me.

Comment (justinhynes) [reply requirements/base.txt]: running `make upgrade` reverted my change back to the original version, so I'll roll with that.

**Pull request merged**